src/Board.hs

```diff
@@ -101,9 +103,8 @@ isStartingLayout board = board == calculateStartingBoard board
 -- Return what a given board would look like in its unscrambled, initial
 -- layout.
 calculateStartingBoard :: Board -> Board
-calculateStartingBoard (Board d _) =
-    Board d $ generateOrderedLayout d
-    where generateOrderedLayout (Dimensions width height) =
-              let lastNumber = width * height - 1
-               in (map Just [1 .. lastNumber]) ++ [Nothing]
-
+calculateStartingBoard (Board d _) = Board d (generateOrderedLayout d)
+  where
+    generateOrderedLayout (Dimensions width height) =
+      let lastNumber = width * height - 1
+      in  (map Just [1 .. lastNumber]) ++ [Nothing]
```

Comment:
I just noticed a style point here... Is two-space indention community standard in Haskell?